ci/matrix.sh

```diff
@@ -64,7 +64,7 @@ elif [ "$BUILD_TARGET" = "linux64" ]; then
   export HOST=x86_64-unknown-linux-gnu
   export PACKAGES="bc python3-zmq"
   export DEP_OPTS="NO_UPNP=1 DEBUG=1"
-  export BITCOIN_CONFIG="--enable-zmq --enable-glibc-back-compat --enable-reduce-exports --enable-stacktraces"
+  export BITCOIN_CONFIG="--enable-zmq --enable-glibc-back-compat --enable-reduce-exports --enable-stacktraces --enable-statsd"
```

Comment:
I would strongly prefer to NOT make statsd support dependent on how you compile dash. We have everything that we need for this inside the Dash source, so there are no external dependencies that might require us to make it conditional.
If this is about performance/overhead concerns, then it would be enough to add a method (e.g. `statsClient.IsEnabled()`) that allows to check for it at runtime, so that we can enclose expensive measurements with an if (e.g. the ones that do string concatenation). That same method should then be checked internally in `StatsdClient` in every method so that we bail out early.

Reply:
There is `-statsenabled` cmd line option which disables this functionality by default with minimal overhead 
b9e0373. The idea to not compile it at all is basically the same as for the miner - it's an optional module which has nothing to do with the core node functionality itself. Also, there is no need to include it in windows or mac os builds and someone might want to run a node on linux without it just like someone might want to run a node without wallet support for whatever reason. IMO we should have both - compile time option and cmd line one.

Reply:
There was a good reason for the miner to be compile time dependent, which was the issue with anti virus false positives.

TBH, I would even prefer to not have wallet support compile time dependent. It should be enough to just allow disabling it via command line. But this is something we won't touch due to possible backporting merge conflicts. 

The reason I'm against such compile time switches is because in my opinion it complicates code (all these nasty #ifdef/#endifs) and leads to all kinds of issue which are usually only later discovered. The compile time wallet support shows this again and again where things unexpectedly break because it was not taken into consideration that wallet support must be checked at compile time AND run time.

If someone doesn't want to use statsd, he should simply not use the command line switch. Otherwise, where is the line? If someone doesn't like multisig addresses, why not add a compile time switch to remove that feature from compilation?

IMHO, things like this are a typical C/C++ thing and the only reason why it should be done is to allow compilation for more targets. For example, when dependencies are not easily satisfied on all targets or when something is not implementable on ARM or Windows for example. This is the only time when I believe that the code cluttering is worth it.

I mean...I won't go crazy if you still say that you want to keep it, but I guess I'm kind of suffering from some mild form of PTSD due to projects I've been working on years ago :see_no_evil: :D

Reply:
Hmm.. ok, you convinced me :)